### PR TITLE
Add New Package animation-utils

### PR DESCRIPTION
**Pull request by ChetanKarwa**

### CODE REVIEW

File: packages/docs/docs/animation-utils/use-interpolate-styles.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+```tsx twoslash
+// ---cut---
+import {
+  useInterpolateStyles,
+  makeTransform,
+  translateY,
+} from "@remotion/animation-utils";
+
+const MyComponent: React.FC = () => {
+  const animatedStyles = useInterpolateStyles({
```
Comment (JonnyBurger) [inline packages/docs/docs/animation-utils/use-interpolate-styles.md]: It's a nice idea! But this makes it less flexible unfortunately: You have to obey by the rules of hooks (no conditional rendering, cannot .map() over them), and there is no control over the timing, everything seems to be linear timing. You also cannot use it outside a component or with a spring animation. The API should work more like interpolate:

```tsx
const styles = interpolateStyles(frame, [0, 30, 60], [{opacity: 0}, {opacity: 1}, {opacity: 0}])
```

So it doesn't need to be a hook.

File: packages/docs/docs/animation-utils/make-transform.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+| **translate(x, y = 0, unitX = 'px', unitY = unitX)** | Translates an element in the 2D plane.                       |
+| **translate3d(x, y, z, unitX = 'px', ...)**          | Translates an element in 3D space.                           |
+| **translateX(x, unit = 'px')**                       | Translates an element along the horizontal axis in 2D space. |
+| **translateY(y, unit = 'px')**                       | Translates an element along the vertical axis in 2D space.   |
+| **translateZ(z, unit = 'px')**                       | Translates an element along the z-axis in 3D space.          |
+
+## Usage
+
+```typescript
+import { makeTransform, rotate, translate } from "@remotion/animation-utils";
```
Comment (JonnyBurger) [inline packages/docs/docs/animation-utils/make-transform.md]: Can we add twoslash type checking for this and putt he example on top of the page?

File: packages/docs/docs/animation-utils/use-interpolate-styles.md (hunk 1 above)
Comment (JonnyBurger) [reply packages/docs/docs/animation-utils/use-interpolate-styles.md]: It also means the function cannot be used outside of Remotion, which could be useful. I'd like if we scrap this function entirely and just ship a "pure" function

File: packages/docs/docs/animation-utils/interpolate-styles.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+```tsx twoslash
+// ---cut---
+import {
+  interpolateStyles,
+  makeTransform,
+  translateY,
+} from "@remotion/animation-utils";
+
+const MyComponent: React.FC = () => {
+  const animatedStyles = interpolateStyles({
```
Comment (JonnyBurger) [inline packages/docs/docs/animation-utils/interpolate-styles.md]: The function signature is wrong (this is also why the Vercel deployment is failing)

File: packages/animation-utils/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
+# `@remotion/animation-utils`
+
+## interpolateStyles
+
+_Available from v2.0.3_
```
Comment (JonnyBurger) [inline packages/animation-utils/README.md]: Can you put 4.0.89?

File: packages/animation-utils/README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+				Testing
+			</h1>
+		</AbsoluteFill>
+	);
+}
+```
+
+## makeTransform
+
+A type-safe function to create string for the `transform` CSS property.
```
Comment (JonnyBurger) [inline packages/animation-utils/README.md]:
```suggestion
A type-safe function to create a string to be passed to the `transform` CSS property.
```

File: packages/animation-utils/package.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
+{
+	"name": "@remotion/animation-utils",
+	"author": {
+		"name": "Chetan Karwa",
+		"email": "cbkarwa@gmail.com"
+	},
+	"version": "4.0.86",
```
Comment (JonnyBurger) [inline packages/animation-utils/package.json]:
```suggestion
	"version": "4.0.88",
```

File: packages/animation-utils/src/transformation-helpers/interpolate-styles/index.tsx [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+	for (const index in arr) {
+		if (typeof arr[index] !== 'object') {
+			throw new Error('outputStyles must contain only objects');
+		}
+	}
+}
+
+/**
+ * @description This function allows you to map a range of values to colors using a concise syntax.
+ * @see [Documentation](https://www.remotion.dev/docs/interpolate-colors)
```
Comment (JonnyBurger) [inline packages/animation-utils/src/transformation-helpers/interpolate-styles/index.tsx]: This is inaccurate

File: packages/docs/components/TableOfContents/animation-utils.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import { TOCItem } from "./TOCItem";
+
+export const TableOfContents: React.FC = () => {
+  return (
+    <div>
+      <Grid>
+        <TOCItem link="/docs/animation-utils/make-transform">
+          <strong>makeTransform()</strong>
+          <div>
+            Type-safe function to create string for the transform CSS property
```
Comment (JonnyBurger) [inline packages/docs/components/TableOfContents/animation-utils.tsx]:
```suggestion
            Typesafe function to create a string for the CSS <code>transform</code> property
```

File: packages/docs/components/TableOfContents/animation-utils.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+          </div>
+        </TOCItem>
+        <TOCItem link="/docs/animation-utils/interpolate-styles">
+          <strong>interpolateStyles()</strong>
+          <div>Map a range of values to Styles</div>
```
Comment (JonnyBurger) [inline packages/docs/components/TableOfContents/animation-utils.tsx]:
```suggestion
          <div>Map a range of values to CSS <code>style</code> values</div>
```

File: packages/docs/components/TableOfContents/api.tsx [hunk 10]
```diff
@@ -96,6 +100,10 @@ export const TableOfContents: React.FC = () => {
           <strong>@remotion/layout-utils</strong>
           <div>Layout helpers</div>
         </TOCItem>
+        <TOCItem link="/docs/animation-utils">
+          <strong>@remotion/animation-utils</strong>
+          <div>Keyframing utilities</div>
+        </TOCItem>
```
Comment (JonnyBurger) [inline packages/docs/components/TableOfContents/api.tsx]: Oops sorry I duplicated it

```suggestion
```

File: packages/docs/docs/animation-utils/index.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
+---
+image: /generated/articles-docs-animation-utils-index.png
+title: "@remotion/animation-utils"
+---
+
+_Available from v4.0.86_
```
Comment (JonnyBurger) [inline packages/docs/docs/animation-utils/index.md]:
```suggestion
_Available from v4.0.89_
```

File: packages/animation-utils/README.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+			</h1>
+		</AbsoluteFill>
+	);
+}
+```
+
+## License
+
+See the [LICENSE.md](LICENSE.md) file for the license of this repo.  
+To use Remotion, a company license is needed for some entities. [Read the terms here](https://github.com/remotion-dev/remotion/blob/main/LICENSE.md).
```
Comment (JonnyBurger) [inline packages/animation-utils/README.md]:
```suggestion
MIT
```

**Review by JonnyBurger — CHANGES_REQUESTED** [review]

File: packages/animation-utils/src/transformation-helpers/interpolate-styles/index.tsx (hunk 7 above)
Comment (ChetanKarwa) [reply packages/animation-utils/src/transformation-helpers/interpolate-styles/index.tsx]: By mistake copied exactly was taking insperation from interpolate color function 😅

File: packages/docs/components/TableOfContents/api.tsx (hunk 10 above)
Comment (ChetanKarwa) [reply packages/docs/components/TableOfContents/api.tsx]: Should I use `Keyframing Utilities` or `Utilities for animation` is fine?

**Review by JonnyBurger — APPROVED** [review]

**Pull request merged**